catroid/src/main/java/org/catrobat/catroid/content/actions/DronePlayLedAnimationAction.java

```diff
@@ -26,16 +26,132 @@
 import com.parrot.freeflight.drone.DroneProxy.ARDRONE_LED_ANIMATION;
 import com.parrot.freeflight.service.DroneControlService;
 
+import org.catrobat.catroid.content.bricks.DronePlayLedAnimationBrick;
 import org.catrobat.catroid.drone.ardrone.DroneServiceWrapper;
 
 public class DronePlayLedAnimationAction extends TemporalAction {
+	private DroneControlService dcs;
+	private DronePlayLedAnimationBrick.LedAnimation ledAnimationType;
+	private float duration = 5.0f;
+
+	public void setAnimationType(DronePlayLedAnimationBrick.LedAnimation ledAnimationType) {
+		this.ledAnimationType = ledAnimationType;
+	}
 
 	@Override
 	protected void begin() {
 		super.begin();
-		DroneControlService dcs = DroneServiceWrapper.getInstance().getDroneService();
+		dcs = DroneServiceWrapper.getInstance().getDroneService();
+		if (dcs != null) {
+			ledAnimation();
+		}
+	}
+
+	protected void ledAnimation() {
 		if (dcs != null) {
-			dcs.playLedAnimation(5.0f, 3, ARDRONE_LED_ANIMATION.ARDRONE_LED_ANIMATION_BLINK_ORANGE.ordinal());
+			switch (ledAnimationType) {
+				case STANDARD:
```

Comment:
since this is just mapping values of a LedAnimation enum to some ARDRONE_LED.ARDRONE_LED_ANIMATION_STANDARD wouldnt it be possible to use the enum from ardrone and not have our own?

or if above is not wished, create a mapping of those in a hashmap:
HashMap<LedAnimation, DroneProxy.ARDRONE_LED_ANIMATION> animationTypeMapping = new HashMap<>();
animationTypeMapping.put(LedAnimation.BLANK, DroneProxy.ARDRONE_LED_ANIMATION.ARDRONE_LED_ANIMATION_BLANK);

or at least put 
dcs.playLedAnimation(5.0f, (int) duration, ARDRONE_LED_ANIMATION.ARDRONE_LED_ANIMATION_BLANK.ordinal());
super.setDuration(duration);
in a function with the ARDRONE_LED_ANIMATION as parameter

with any of the above the size of this would shrink from 100 lines chunk of code to sub 50...

Reply:
thanks for the helpful hints, i will implement a shorter and better solution for this different drone animation types